server/webapp/WEB-INF/rails.new/app/controllers/api_v1/version_infos_controller.rb

```diff
@@ -0,0 +1,53 @@
+##########################################################################
+# Copyright 2015 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+module ApiV1
+  class VersionInfosController < ApiV1::BaseController
+    def update_server
+      result = HttpLocalizedOperationResult.new
+
+      render :nothing => true, :status => 400 and return if message_tampered?
+
+      message = JSON.parse(params['message'])
+      version_info = version_info_service.updateServerLatestVersion(message['latest-version'], result)
+
+      if result.httpCode == 200
```

Comment:
Also, I've been thinking, we've used this pattern in a few places. Does this call for a prettier way to write this (just thinking out aloud). This need not be done right now.

```ruby
return if_success(result) do
  render :json_hal_v1 => to_json_hal(version_info)
end


def if_success(result, &blk)
  if result.isSuccessful?
    yield
    true
  else
    render_http_operation_result(result)
    false
  end
end
```